dropshots-gradle-plugin/src/main/kotlin/com/dropbox/dropshots/device/DeviceProviderFactory.kt

```diff
@@ -0,0 +1,30 @@
+package com.dropbox.dropshots.device
+
+import com.android.build.gradle.internal.LoggerWrapper
+import com.android.build.gradle.internal.testing.ConnectedDeviceProvider
+import com.android.builder.testing.api.DeviceProvider
+import com.android.ddmlib.DdmPreferences
+import com.android.utils.ILogger
+import java.io.File
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Internal
+
+public abstract class DeviceProviderFactory {
+  @Suppress("DEPRECATION")
+  @get:Internal
+  internal var deviceProvider: DeviceProvider? = null
```

Comment:
Is this assigned anywhere? What's the reason for using a Nested Bean instead of just some helper method for `getDeviceProvider`?

Reply:
This is [modeled after AGP](https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-core/src/main/java/com/android/build/gradle/internal/tasks/DeviceProviderInstrumentTestTask.java;l=261?q=DeviceProviderInstrumentTestTask&ss=android-studio%2Fplatform%2Ftools%2Fbase), and I left this as a provider to enable tests that don't require connecting to actual devices. Those tests aren't implemented, and may not be, but I left this to leave that possibility.